spec/unit/models/runtime/v3/persistence/isolation_segment_model_spec.rb

```diff
@@ -2,81 +2,210 @@
+
+              expect(isolation_segment_model.organizations).to eq([org])
+              expect(org.isolation_segment_models).to eq([isolation_segment_model])
+              expect(org.isolation_segment_model).to eq(isolation_segment_model)
+            end
+
+            context 'and the isolation segment has been added to a space in the org' do
+              let!(:space) { Space.make(organization: org, isolation_segment_guid: isolation_segment_model.guid) }
+
+              context 'and we remove the isolation segment' do
+                it 'does not allow the isolation segment to be deleted' do
+                  expect {
+                    isolation_segment_model.remove_organization(org)
+                  }.to raise_error(CloudController::Errors::ApiError)
+
+                  expect(isolation_segment_model.organizations).to eq([org])
+                  expect(org.isolation_segment_models).to eq([isolation_segment_model])
+                  expect(org.isolation_segment_model).to eq(isolation_segment_model)
+                end
+              end
+            end
+          end
+
+          context 'and multiple isolation segments are in allowed list' do
+            let(:isolation_segment_model_2) { IsolationSegmentModel.make }
+
+            before do
+              isolation_segment_model.add_organization(org)
+              isolation_segment_model_2.add_organization(org)
+            end
+
+            it 'cannot remove the isolation segment that has been set as the default' do
+              expect(org.isolation_segment_model).to eq(isolation_segment_model)
+              expect {
+                isolation_segment_model.remove_organization(org)
+              }.to raise_error(CloudController::Errors::ApiError)
+            end
+
+            it 'can remove an isolation segment that is not the default' do
+              expect(org.isolation_segment_model).to_not eq(isolation_segment_model_2)
+              isolation_segment_model_2.remove_organization(org)
+              expect(org.isolation_segment_models).to eq([isolation_segment_model])
+            end
+
+            context 'and an isolation segment that is not the default has been associated with a space' do
+              let(:space) { Space.make(organization: org) }
+
+              before do
+                expect(org.isolation_segment_model).to_not eq(isolation_segment_model_2)
+                space.isolation_segment_model = isolation_segment_model_2
+                space.save
+              end
+
+              it 'does not allow the isolation segment to be deleted' do
+                expect {
+                  isolation_segment_model_2.remove_organization(org)
+                }.to raise_error(CloudController::Errors::ApiError)
+              end
+            end
+          end
+        end
+
+        context 'when setting the default isolation segment' do
+          it 'must be in the allowed list' do
```

Comment:
Ugh.  This behavior is partially handled (where it should be) in the organization model spec, but there are empty tests there as well.   Fixing.